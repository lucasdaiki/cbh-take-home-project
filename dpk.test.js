const { deterministicPartitionKey } = require("./dpk");

describe("deterministicPartitionKey", () => {
  it("Returns the literal '0' when given no input", () => {
    const partitionKey = deterministicPartitionKey();
    expect(partitionKey).toBe("0");
  });

  it("Returns the partitionKey (string) when partition key is provided", () => {
    const event = { partitionKey: "testPartitionKey" };

    const partitionKey = deterministicPartitionKey(event);
    expect(partitionKey).toBe("testPartitionKey");
  });

  it("Returns the partitionKey (json) when partition key is provided", () => {
    const event = { partitionKey: { test: "PartitionKey" } };

    const partitionKey = deterministicPartitionKey(event);
    expect(partitionKey).toBe(JSON.stringify(event.partitionKey));
  });

  it("Returns a hash when partition key is not provided", () => {
    const event = { anotherField: { partition: "key" } };

    const partitionKey = deterministicPartitionKey(event);
    expect(partitionKey).toMatchSnapshot();
  });

  it("Returns a hash when candidate is bigger than the MAX_PARTITION_KEY_LENGTH", () => {
    const event = { partitionKey: new Array(300).fill('a').join('') };

    const partitionKey = deterministicPartitionKey(event);
    expect(partitionKey).toMatchSnapshot();
  });
});
