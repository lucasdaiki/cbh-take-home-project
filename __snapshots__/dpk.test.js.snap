// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deterministicPartitionKey Returns a hash when partition key is not provided 1`] = `"de8bb975b898b524a6dc641acac54df5dbbd84320e8d87bca24098f17c754a20d3d7b6d131969cef58be21577a5833debbe8f673303649ee44b34b26f119909b"`;
